Use xpc_dictionary_create_empty to create empty dictionaries

AZero13 opened the pull request:
xpc_dictionary_create has Nonnull annotated for some of its parameters. We should just do xpc_dictionary_create_empty to avoid warnings and be safe.

<!--
If this pull request is targeting a release branch, please fill out the
following form:
https://github.com/swiftlang/.github/blob/main/PULL_REQUEST_TEMPLATE/release.md?plain=1

Otherwise, replace this comment with a description of your changes and
rationale. Provide links to external references/discussions if appropriate.
If this pull request resolves any GitHub issues, link them like so:

  Resolves <link to issue>, resolves <link to another issue>.

For more information about linking a pull request to an issue, see:
https://docs.github.com/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

bnbarham commented:
Thanks @RSilicon! Presumably this doesn't actually matter because of the 0 count, but all for removing warnings. We also just bumped our minimum deployment to 13 (< 11) before, so can indeed use this now.

Note that there's still a few more 😅:
```
tools/sourcekitd/bin/XPC/Service/XPCService.cpp
49:    xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr, 0); // missing
93:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr,  0);
130:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr,  0);
331:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr,  0);
```

```
tools/sourcekitd/bin/XPC/Client/sourcekitd.cpp
136:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr,  0);
185:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr,  0);
221:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr, 0); // missing
232:  xpc_object_t msg = xpc_dictionary_create(nullptr, nullptr, 0); // missing
370:  xpc_object_t ping_msg = xpc_dictionary_create(nullptr, nullptr, 0); // missing
```

```
tools/sourcekitd/lib/API/sourcekitdAPI-XPC.cpp
169:  Impl = xpc_dictionary_create(nullptr, nullptr, 0); // missing
266:  xpc_object_t dict = xpc_dictionary_create(nullptr, nullptr, 0); // missing
281:  xpc_object_t dict = xpc_dictionary_create(nullptr, nullptr, 0); // missing
```

AZero13 committed:
Use xpc_dictionary_create_empty to create empty dictionaries

xpc_dictionary_create has Nonnull annotated for some of its parameters. We should just do xpc_dictionary_create_empty to avoid warnings and be safe.

bnbarham commented:
@swift-ci please test

bnbarham reviewed (approved):
Failures are unrelated @RSilicon. I'll run the tests again once PR testing is green.

Thanks for fixing up `xpc_array_create` as well!

bnbarham commented:
@swift-ci please test

bnbarham commented:
Thanks @AreaZR!
